### PR TITLE
Add ALTER TYPE statements for postgres

### PR DESCRIPTION
**Pull request by MozarellaMan**
For #38 

This PR isn't ready. This is a draft so I can check if I am doing things right. 

I have implemented ADD VALUE and RENAME TO so far. Before I start on OWNER, SET SCHEMA, ADD ATTRIBUTE, and the RESTRICT and CASCADE options - I wanted to some feedback to see if I'm on the right track with the builder and sql generation code.

If everything is fine I will continue with the PR and add documentation, if I'm doing anything wrong or there is a better way let me know!

### CODE REVIEW

File: src/extension/postgres/types.rs [hunk 1]
```diff
@@ -28,18 +28,54 @@ pub struct TypeDropStatement {
...
+        create: &TypeCreateStatement,
+        sql: &mut SqlWriter,
+        collector: &mut dyn FnMut(Value),
+    );
 
     /// Translate [`TypeDropStatement`] into database specific SQL statement.
-    fn prepare_type_drop_statement(&self, create: &TypeDropStatement, sql: &mut SqlWriter, collector: &mut dyn FnMut(Value));
+    fn prepare_type_drop_statement(
+        &self,
+        create: &TypeDropStatement,
```
Comment (billy1624) [inline src/extension/postgres/types.rs:67]: Just a minor typo. Should be `drop`?

**Comment by tyt2y3** [comment]
Thank you for the PR, nicely done! If you intend to format an existing piece of code, please do so in a standalone `Format code` commit, such that feature additions will not be mixed together.

File: src/extension/postgres/types.rs (hunk 1 above)
Comment (MozarellaMan) [reply src/extension/postgres/types.rs:67]: Sorry, what are you referring to?

**Comment by MozarellaMan** [comment]
> 
> 
> Thank you for the PR, nicely done! If you intend to format an existing piece of code, please do so in a standalone `Format code` commit, such that feature additions will not be mixed together.

Yep, that makes sense. Will avoid just running `cargo fmt` in future. I've added the  `RENAME VALUE` statement as well. I just realised I accidentally PR'd from my fork's master instead of the `add_alter_type` feature branch, sorry about that!

File: src/extension/postgres/types.rs (hunk 1 above)
Comment (billy1624) [reply src/extension/postgres/types.rs:68]:
```suggestion
        drop: &TypeDropStatement,
```

**Comment by tyt2y3** [comment]
```
---- src/extension/postgres/types.rs - extension::postgres::types::TypeAlterStatement::name (line 308) stdout ----
Test executable failed (exit code 101).

stderr:
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"ALTER TYPE \"font_family\" ADD VALUE \'cursive\'"`,
 right: `"ALTER TYPE \"font_family\" ADD VALUE \'cursive\' "`', src/extension/postgres/types.rs:24:1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
The test fails because of the extra space at the end of the assert.
You can run all test cases (including doc tests by running)
```
cargo test --doc --all-features
```

**Comment by MozarellaMan** [comment]
I missed that! Thanks

**Comment by tyt2y3** [comment]
Looks good to me! Ready for merging?

**Comment by tyt2y3** [comment]
I intentionally ran cargo fmt on my branch to 'subtract out' those lines.

**Pull request merged**